include/systems/eigen_system.h

```diff
@@ -273,6 +307,11 @@ class EigenSystem : public System
    * A boolean flag to indicate whether or not to use a shell preconditioning matrix
    */
   bool _use_shell_precond_matrix;
+
+  /**
+   * Some systems need an arbitrary number of matrices.
+   */
+  std::map<std::string, Matrix *> _matrices;
```

Comment:
You build() and release() matrices into `_matrices` elsewhere in this PR, yes? Are they cleaned up somewhere?